### ISSUE TITLE
IGDB says key is unauthorized (discontinued v3 API) [FIXED]

### ISSUE THREAD

**Issue by hacooke** [muldjord/skyscraper#266]
**Describe the bug**
I have been unable to get skyscraper running with igdb as the scraping module. I have asked on the igdb discord and they are in the process of discontinuing the version of the API used in the skyscraper code (v3). I'm not sure if it is currently still working for people with existing API keys, but I seem to only be able to get a key for the new v4, which has a new authentication system.

**To Reproduce**
Run skyscraper with the `-s igdb` option. If it works as it is, try getting a new API key following the documentation linked from the skyscraper docs and then run again.

**Special circumstances**
Discontinuation of IGDB API v3

**Technical information**
 - Skyscraper version: 3.5.8
 - Internet connection: wifi
 - OS: Ubuntu
 - OS Version: 16.04.2 LTS

**Comment by muldjord** [comment]
Hi, yes, this is correct. I need to update the IGDB authentication. I'll let this stay open so people can see I am aware of the issue.

**Comment by muldjord** [comment]
Good news, I decided to look into this and I now have a working version on master. I am a bit unsure whether the request limits per second they now have are per app ID, or if it's per computer it is run on. I will need to read up on that. But yeah, implementing it was easier than I anticipated.

**Comment by hacooke** [comment]
Awesome! I'll check it out (if you'll pardon the outdated version-control pun)

**Comment by muldjord** [comment]
Please wait for a bit, I am improving it a bit at the moment. I'll let you know when it's optimized (won't be long).

**Comment by muldjord** [comment]
I've had a chat with the IGDB guys on Discord, and they seem to allow users to create their own credentials for use with the IGDB module. This is good news, as this allows users to get their own "request limits".

If you have the time to test this, please read more about how to get your own IGDB credentials here: https://github.com/muldjord/skyscraper/blob/master/docs/SCRAPINGMODULES.md#igdb

I am very interested in hearing if it works with your credentials as this is something that's hard for me to test myself as I only have my own which are already included with Skyscraper.

**Comment by muldjord** [comment]
I just found a bug, so don't bother testing right now.

**Comment by muldjord** [comment]
So, I finally got some time to look more into this today. I had a chat with the good guys at IGDB over Discord, and the implementation is now described here: https://github.com/muldjord/skyscraper/blob/master/docs/SCRAPINGMODULES.md#igdb

If you are able to test it, I would greatly appreciate it (it's on 'master'). As the link says, you need a personal key/secret to use it. Let me know if you run into issues.

**Comment by muldjord** [comment]
After some further testing I've decided to release. It's now in the just-released 3.5.9. Let me know if you have any issues.

**Comment by hacooke** [comment]
It seems I get some errors when trying to install 3.5.9:
```C++
src/skyscraper.cpp: In member function ‘void Skyscraper::doPrescrapeJobs()’:
src/skyscraper.cpp:1657:69: error: ‘currentSecsSinceEpoch’ is not a member of ‘QDateTime’
     qlonglong tokenLife = tokenData.split(';').at(2).toLongLong() - QDateTime::currentSecsSinceEpoch();
                                                                     ^
src/skyscraper.cpp:1674:14: error: ‘currentSecsSinceEpoch’ is not a member of ‘QDateTime’
  tokenLife = QDateTime::currentSecsSinceEpoch() + jsonObj["expires_in"].toInt();
              ^
src/skyscraper.cpp:1676:102: error: ‘currentSecsSinceEpoch’ is not a member of ‘QDateTime’
    tokenFile.write(config.user.toUtf8() + ";" + config.igdbToken.toUtf8() + ";" + QByteArray::number(QDateTime::currentSecsSinceEpoch() + tokenLife));
                                                                                                      ^
```

**Comment by muldjord** [comment]
Ah, that's because the function 'currentSecsSinceEpoch' wasn't introduced until Qt 5.8. I've now released Skyscraper 3.6.0 where it uses 'currentMSecsSinceEpoch' instead which has been supported all the way back to Qt 4.7. Please update and thanks for reporting it.
